drivers/dac/dac_dacx3608.c

```diff
@@ -0,0 +1,282 @@
+	if (ret) {
+		return -EIO;
+	}
+
+	if (setting) {
+		regval |= mask;
+	} else {
+		regval &= ~mask;
+	}
+
+	ret = dacx3608_reg_write(dev, reg, regval);
+	if (ret) {
+		return ret;
+	}
+
+	return 0;
+}
+
+static int dacx3608_channel_setup(const struct device *dev,
+				   const struct dac_channel_cfg *channel_cfg)
+{
+	const struct dacx3608_config *config = dev->config;
+	struct dacx3608_data *data = dev->data;
+	bool setting = false;
+	int ret;
+
+	if (channel_cfg->channel_id > DACX3608_MAX_CHANNEL - 1) {
+		LOG_ERR("Unsupported channel %d", channel_cfg->channel_id);
+		return -ENOTSUP;
+	}
+
+	if (channel_cfg->resolution != config->resolution) {
+		LOG_ERR("Unsupported resolution %d", channel_cfg->resolution);
+		return -ENOTSUP;
+	}
+
+	if (data->configured & BIT(channel_cfg->channel_id)) {
+		LOG_DBG("Channel %d already configured", channel_cfg->channel_id);
+		return 0;
+	}
+
+	/* Clear PDNn bit */
+	ret = dacx3608_reg_update(dev, DACX3608_REG_DEVICE_CONFIG,
+				BIT(channel_cfg->channel_id), setting);
+	if (ret) {
+		LOG_ERR("Unable to update DEVICE_CONFIG register");
+		return -EIO;
+	}
+
+	data->configured |= BIT(channel_cfg->channel_id);
+
+	LOG_DBG("Channel %d initialized", channel_cfg->channel_id);
+
+	return 0;
+}
+
+static int dacx3608_write_value(const struct device *dev, uint8_t channel,
+				uint32_t value)
+{
+	const struct dacx3608_config *config = dev->config;
+	struct dacx3608_data *data = dev->data;
+	uint16_t regval;
+	int ret;
+
+	if (channel > DACX3608_MAX_CHANNEL - 1) {
+		LOG_ERR("Unsupported channel %d", channel);
+		return -ENOTSUP;
+	}
+
+	if (!(data->configured & BIT(channel))) {
+		LOG_ERR("Channel %d not initialized", channel);
+		return -EINVAL;
+	}
+
+	if (value >= (1 << (config->resolution))) {
+		LOG_ERR("Value %d out of range", value);
+		return -EINVAL;
+	}
+
+	/*
+	 * Shift passed value two times left because first two bits are Don't Care
+	 *
+	 * DACn_DATA register format:
+	 *
+	 * | 15 14 13 12 |      11 10 9 8 7 6 5 4 3 2      |    1 0     |
+	 * |-------------|---------------------------------|------------|
+	 * | Don't Care  |  DAC53608[9:0] / DAC43608[7:0]  | Don't Care |
+	 */
+	regval = value << 2;
+	regval &= 0xFFFF;
+
+	ret = dacx3608_reg_write(dev, DACX3608_REG_DACA_DATA + channel, regval);
+	if (ret) {
+		LOG_ERR("Unable to set value %d on channel %d", value, channel);
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int dacx3608_soft_reset(const struct device *dev)
+{
+	uint16_t regval = DACX3608_SW_RST;
+	int ret;
+
+	ret = dacx3608_reg_write(dev, DACX3608_REG_STATUS_TRIGGER, regval);
+	if (ret) {
+		return -EIO;
+	}
+	k_msleep(DACX3608_POR_DELAY);
+
+	return 0;
+}
+
+static int dacx3608_device_id_check(const struct device *dev)
+{
+	struct dacx3608_data *data = dev->data;
+	int ret;
+
+	ret = dacx3608_reg_read(dev, DACX3608_REG_STATUS_TRIGGER, &data->dev_id);
+	if (ret) {
+		LOG_ERR("Unable to read device ID");
+		return -EIO;
+	}
+
+	switch (data->dev_id) {
```

Comment:
`dev_id` is used only for verification in this function.
Is there any reason this information should be included in the `struct dacx3608_data`?

Reply:
Not really, it can be moved to temp variable in this function.

Reply:
I think it's better to do it as a temp variable inside a function.

Reply:
Fixed.